pyflac/decoder.py

```diff
@@ -314,6 +314,14 @@ def _read_callback(_decoder,
     If an exception is raised here, the abort status is returned.
     """
     decoder = _ffi.from_handle(client_data)
+
+    while len(decoder._buffer) == 0 and not (decoder._error or decoder._done):
+        # ----------------------------------------------------------
+        # Wait until there is something in the buffer, or an error
+        # occurs, or the end of the stream is reached.
+        # ----------------------------------------------------------
+        time.sleep(0.01)
```

Comment:
What would happen if some other part of the system failed while the buffer length is 0 -- is it guaranteed that decoder._error or decoder._done will be set?

Reply:
The `_error_callback` should handle all failures from the FLAC library. It is not guaranteed that `_done` will be set, it is up to the user to call `finish`. I updated the comment in the `finish` method to highlight this.